### PR TITLE
Secrets

### PR DESCRIPTION
**Pull request by mairh**
- The idea behind this PR is to secure important details, which shouldn't go in the code repo.
- People can easily add `secrets.js` in `.gitignore`. All the password and sensitive info are in one place rather all over config folder files.
- I didn't add the `secrets.js` in `.gitignore` in this PR because we don't have any sensitive data.

P.S -> By default the verification is enabled which can be accessed at
https://ethereal.email/

Username and password are in `.env` file

### CODE REVIEW

**Comment by vlasenko** [comment]
I would rather use .env file for secrets

**Comment by mitjade** [comment]
+1 for .env

Otherwise I agree this needs to be extracted from configuration files. Also `googleAnalytics` from `app.json`.

**Comment by mairh** [comment]
Would it directly be available from `process.env.xxxx`?

Can I simply replace

```
export default {
  service: config.mailer.emailService,
  auth: {
    user: config.mailer.emailUser,
    pass: config.mailer.emailPassword
  }
};
```

with

```
export default {
  service: process.env.EMAIL_SERVICE,
  auth: {
    user: process.env.EMAIL_USER,
    pass: process.env.EMAIL_PASSWORD
  }
};
```

and in `.env``

```
EMAIL_SERVICE=Gmail
EMAIL_USER=something@gmai.com
EMAIL_PASSWORD=somePassword
```

**Comment by vlasenko** [comment]
@mairh Yes

**Comment by mairh** [comment]
@vlasenko @mitjade Should be ready to go now from my side

**Comment by mitjade** [comment]
Could you also add `googleAnalytics` from `app.json`. Might be a problem because `app.json` is not `.js`?

**Comment by mairh** [comment]
I've added the tracking io .env file

File: .gitignore [hunk 1]
```diff
@@ -6,3 +6,4 @@ npm-debug.*
 .expo/
 *.sqlite3
 .DS_Store
+jsconfig.json
```
Comment (mitjade) [inline .gitignore]: Do we still need this ignore?

Comment (mairh) [reply .gitignore]: This is not related to this PR. But in some IDE's (for e.g. VS Code) if you have enabled `jshint` extension this file is added by default to the root folder. This is just for safe keeping

Comment (mitjade) [reply .gitignore]: OK

**Comment by mairh** [comment]
@vlasenko I will merge this since it is blocking @mitjade FB auth PR. If you have anything to add pls let me know and I will change that in another PR.

**Comment by mairh** [comment]
@mitjade Looks like the `.env` thing is not working. I only tested with the `secrets.js` but didn't test the application when I switched to `.env`. Not sure why the config can't fetch the values from `.env`

**Comment by mitjade** [comment]
@mairh You might need to add https://github.com/motdotla/dotenv.

**Comment by mitjade** [comment]
This might not work for `GOOGLE_TRACKING_ID`.

**Comment by mairh** [comment]
If we add `dotenv` we need to change webpack config in spinjs to pass

```
node: {
  fs: 'empty'
}
```
We were already using `APP_DIR` from `.env` in Docker files. I wonder do we really need `dotenv`?

**Comment by mitjade** [comment]
@mairh I guess we need to wait for @vlasenko to help out here.

**Comment by vlasenko** [comment]
Let me take a look at this one. Though I didn't understand what should I check, where is the problem?

**Comment by vlasenko** [comment]
Okay, I understood

**Comment by mairh** [comment]
@vlasenko @mitjade This is fixed now via https://github.com/sysgears/apollo-universal-starter-kit/commit/bdfd987ab55686680c741daa8e044a69e1a812e9

**Comment by vlasenko** [comment]
Great! Many thanks @mairh !

**Pull request merged**